### PR TITLE
Add support for extensions and render options

### PR DESCRIPTION
**Pull request by rogerhu**
Add tests to showcase the changes

### CODE REVIEW

File: spec/commonmarker_rouge_spec.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
   end
 
   it 'escapes language markup' do
-    html = CommonMarker::Rouge.render_html(<<-MD)
+    html = CommonMarker::Rouge.render_html(<<-MD, [:DEFAULT], [:UNSAFE])
```
Comment (rogerhu) [inline spec/commonmarker_rouge_spec.rb:21]: this test never worked without the UNSAFE extension?

Comment (arokettu) [reply spec/commonmarker_rouge_spec.rb:21]: I didn't know about the safe flag :D Or maybe there even wasn't one at the moment. It's better to pass :SAFE explicitly (or require 0.18), remove our safety check that seems useless now and update tests

Tests were passing until SAFE became a default (a month ago) The scheduled build was only a week ago and I had no time to address the issue

**Comment by arokettu** [comment]
:+1:

**Comment by arokettu** [comment]
I can merge that at this state or if you also would like to fix that SAFE/UNSAFE issue i can keep it open for a while

File: spec/commonmarker_rouge_spec.rb (hunk 1 above)
Comment (rogerhu) [reply spec/commonmarker_rouge_spec.rb:21]: SAFE got renamed to UNSAFE now...

https://github.com/gjtorikian/commonmarker/commit/1fe234820a77cb3f2b26b47971d229d0da92d652

Are you suggesting we set it explicitly in the actual library?

**Comment by rogerhu** [comment]
Yeah let's merge -- I'm not sure what's the right solution -- force > 0.18 and cause rendering to break for cases when it doesn't work?

**Comment by rogerhu** [comment]
Thanks!  Can you tag and release a new version?  Much appreciated.

**Comment by arokettu** [comment]
@rogerhu I still need to think about #4 because it seems it's not only the test that was broken. I will try to release this weekend

**Comment by rogerhu** [comment]
Regarding your comment in https://github.com/gjtorikian/commonmarker/issues/85#issuecomment-442664355 I modified render_html to allow render options to be passed in....https://github.com/sandfoxme/commonmarker-rouge/pull/3/files#diff-564a9f90cb5bde6b7485e2ac6b3b04e0

```
def render_html(text, cmark_options = :DEFAULT, render_options = :DEFAULT, extensions = [], **cmr_options)
```

**Comment by arokettu** [comment]
@rogerhu I released 1.5.0. It would be very helpful if you also added some usage examples to the readme

**Pull request merged**